File: src-tauri/src/models/domain/timesheet.rs
#[derive(Clone, Copy, Debug, Eq, PartialEq)]
pub enum TimesheetRange {
    Today,
    Week,
    All,
}

#[derive(Clone, Copy, Debug, Eq, PartialEq)]
pub enum TimesheetFormat {
    Full,
    Recent,
}

#[derive(Clone, Copy, Debug, Eq, PartialEq)]
pub struct TimesheetOptions {
    pub range: TimesheetRange,
    pub format: TimesheetFormat,
    pub rounding_enabled: bool,
}

impl TimesheetOptions {
    #[cfg_attr(not(test), allow(dead_code))]
    pub const fn full(range: TimesheetRange) -> Self {
        Self {
            range,
            format: TimesheetFormat::Full,
            rounding_enabled: false,
        }
    }

    #[cfg_attr(not(test), allow(dead_code))]
    pub const fn recent() -> Self {
        Self {
            range: TimesheetRange::Today,
            format: TimesheetFormat::Recent,
            rounding_enabled: false,
        }
    }
}

File: src/models/types/timesheet.ts
export type TimesheetRange = "today" | "week" | "all";
export type TimesheetFormat = "full" | "recent";

export type TimesheetPreviewRow = {
  label: string;
  values: number[];
  total: number;
  is_comment: boolean;
  is_total: boolean;
};

export type TimesheetPreviewSheet = {
  name: string;
  columns: string[];
  rows: TimesheetPreviewRow[];
};

export type TimesheetPreview = {
  title: string;
  generated_at: string;
  generated_at_epoch_ms: number;
  sheets: TimesheetPreviewSheet[];
};

export type TimesheetPreviewRequest = {
  range: TimesheetRange;
  format: TimesheetFormat;
};

export type TimesheetPreviewBootstrap = {
  request: TimesheetPreviewRequest | null;
  rounding_enabled: boolean;
  ui_font_scale: number;
};
